evm-adapters/src/lib.rs

```diff
@@ -179,16 +179,46 @@ pub trait Evm<State> {
 #[cfg(test)]
 mod test_helpers {
     use super::*;
-    use dapp_solc::SolcBuilder;
-    use ethers::{prelude::Lazy, utils::CompiledContract};
+    use ethers::{
+        prelude::Lazy,
+        solc::{
+            artifacts::CompactContract, ArtifactOutput, Project, ProjectCompileOutput,
+            ProjectPathsConfig,
+        },
+    };
     use std::collections::HashMap;
 
-    pub static COMPILED: Lazy<HashMap<String, CompiledContract>> =
-        Lazy::new(|| SolcBuilder::new("./testdata/*.sol", &[], &[]).unwrap().build_all().unwrap());
+    pub static COMPILED: Lazy<HashMap<String, CompactContract>> = Lazy::new(|| {
+        // NB: should we add a test-helper function that makes creating these
+        // ephemeral projects easier?
+        let paths =
+            ProjectPathsConfig::builder().root("testdata").sources("testdata").build().unwrap();
```

Comment:
to make this safe I suggest using `let root = Path::new(env!("CARGO_MANIFEST_DIR")).join("testdata")`.

Ref https://github.com/gakonst/ethers-rs/blob/bfdf70cbea9f96f98cb3ba6bba3db8a335e51aba/examples/contract_with_abi.rs#L21